jupyter-tensorflow-full/rockcraft.yaml

```diff
@@ -0,0 +1,189 @@
+# Based on the following Dockerfiles:
+# https://github.com/kubeflow/kubeflow/blob/v1.7-branch/components/example-notebook-servers/base/Dockerfile
+# https://github.com/kubeflow/kubeflow/blob/v1.7-branch/components/example-notebook-servers/jupyter/Dockerfile
+# https://github.com/kubeflow/kubeflow/blob/v1.7-branch/components/example-notebook-servers/jupyter-tensorflow-full/cpu.Dockerfile
+name: jupyter-tensorflow-full
+summary: An image for using Jupyter & Tensorflow on CPUs
+description: |
+  This image is used as part of the Charmed Kubeflow product. It is deployed
+  in a user namespace, and enables users to perform CPU-accelerated tasks
+  using Tensortflow from within a Jupyter Notebook.
+
+  Both Tensorflow and Jupyter are installed in Conda environment, which is
+  automatically activated.
+version: v1.7.0_20.04_1
+license: Apache-2.0
+base: ubuntu:20.04
+run-user: _daemon_
+services:
+  jupyter:
+    override: replace
+    summary: "jupyter-tensorflow-full service"
+    startup: enabled
+    environment:
+      NB_USER: jovyan
+      NB_UID: 1001
+      NB_PREFIX: /
+      HOME: /home/jovyan
+      SHELL: /bin/bash
+      LANG: en_US.UTF-8
+      LANGUAGE: en_US.UTF-8
+      LC_ALL: en_US.UTF-8
+      PATH: /opt/conda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+      PYTHONPATH: /opt/conda/lib/python3.8/site-packages/
+    command: ./jupyter lab --notebook-dir="/home/jovyan" --ip=0.0.0.0 --no-browser --port=8888 --ServerApp.token="" --ServerApp.password="" --ServerApp.allow_origin="*" --ServerApp.base_url="/" --ServerApp.authenticate_prometheus=False
+    working-dir: /opt/conda/bin/
+platforms:
+  amd64:
+
+parts:
+  security-team-requirement:
+    plugin: nil
+    override-build: |
+      mkdir -p ${CRAFT_PART_INSTALL}/usr/share/rocks
+      (echo "# os-release" && cat /etc/os-release && echo "# dpkg-query" && \
+      dpkg-query -f '${db:Status-Abbrev},${binary:Package},${Version},${source:Package},${Source:Version}\n' -W) \
+      > ${CRAFT_PART_INSTALL}/usr/share/rocks/dpkg.query
+
+  overlay-pkgs:
+    plugin: nil
+    overlay-packages:
```

Comment:
leaving my 2 cents here:
 - beside what @SamirPS already mentioned (wrt many of these utilities already being in the base image), I think you should reconsider this ROCK's purpose. In a simplistic way, you can usually find 3 types of container images: application images (like nginx, prometheus, etc.), distribution images (like ubuntu, alpine), and runtime images (like python, golang). ROCKs are, typically, application images, which means it is weird to find this amount of utilities within. You might have a good reason for having things like `bash`, I won't dispute that, but OTOH, I'd question the inclusion of things like `curl`, `git`, `vim`. Why would these be necessary for an application image? If your answer is: "my users need to be able to debug the image and interact with it", then I'd say: "they can already do that cause your ROCK is based on `ubuntu` and thus already has all the essentials for that. If someone wants an extra utility, like `vim`, they can just `apt install vim`".

Anyway, I might be missing context, but I just wanted to share my view on this cause I feel the resulting ROCK is getting unnecessarily big, complex and prone to vulnerabilities.

Reply:
@cjdcordeiro Thank you for the input. Our Kubeflow ROCKs are strange beasts.
This particular set of images are servers that users interact with and run their own Python scripts with. These are not really an application running by itself, but rather used as base for data scientists to run their own workloads. End users would defintely use `git` and `curl`, `vim` not so much. Some packages could be considered extras, but we would like to keep same user experience as upstream, that's the reason we are keeping all those packages in. We would like end users to not install something that they are used to having in upstream image.

Reply:
ack. Anyway, whenever we have the chance to improve wrt upstream, we should do it (in this case, stuff like `vim` is an obvious candidate to be dropped)